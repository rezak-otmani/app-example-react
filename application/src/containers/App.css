
 .masthead.segment {

     min-height: 700px;
   }
    
   .masthead h1.ui.header {
      margin-top: 0.3em;
      margin-bottom: 0.3em;
      font-size: 6em;
      font-weight: normal;
    
    }

    .masthead h2.ui.header {
      margin-top: 2em;
      margin-bottom: 0em;
      font-size: 1.5em;
      font-weight: normal;
    }



    .masthead .ui.image
     {
      margin-top: 6.1em;
      margin-left: 5.3em;
      margin-bottom: 0em;
     }

     .masthead div.ui.divider {
      width: 15em;
      height: 0.5em;
      line-height: 10em;
      text-align: center;
     }
     .masthead div.embed {
      max-height: 400px;
      height: 0em;
      margin-top: 6em;
      margin-bottom: -1em;
    }

   .headertop
    {

    padding-top: 5em;
    padding-bottom: 0;
    }


    .logo
    {
    width: 15em;
    height: 15em;
    }

    .stylesvgtop
    {
    width:100%;
    height:100%;
    top:3em;
    right:0;
    bottom:0;
    left:0;
    margin:auto;
    fill:#3D9BE9;
    fill-opacity:0;
    stroke:rgba(231, 255, 110, 1);
    stroke-width:0.6em;
    vector-effect:non-scaling-stroke; 
    }

    .stylesvgbottom
    {
    width:100%;
    height:100%;
    top:3em;
    right:0;
    bottom:0;
    left:0;
    margin:auto;
    fill:#3D9BE9;
    fill-opacity:0;
    stroke:rgba(1, 1, 1, 1);
    stroke-width:0.6em;
    vector-effect:non-scaling-stroke;
    }


   .stylesvgtop text
   {
   font-size: 5em;
   font-family: sans-serif;
   text-anchor: middle;
   stroke-width:0.05em;
   }

   .stylesvgbottom text 
   {
   font-size: 5em;
   font-family: sans-serif;
   text-anchor: middle;
   stroke-width:0.05em;
   }

