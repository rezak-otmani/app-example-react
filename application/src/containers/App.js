import React, { Component } from 'react';
import './App.css';
import Home from '../views/home/home';
import About from '../views/about/about';
import Contact from '../views/contact/contact';
import Demo from '../views/demo/demo';
import Featured from '../views/featured/featured';
import Product from '../views/product/product';
import NavBar from '../components/NavBar';


 class App extends Component {

render () {
 return (
  <div class="pusher">
  <div class="ui vertical masthead center aligned segment">
        <div class="ui inverted vertical strip segment"> 
             
          <div class = "headertop">

            <div class="ui three columns center aligned stackable grid">     


<div class="column">
<a href="" target="_self" tabindex="-1">
<svg class = "logo" xmlns="http://www.w3.org/2000/svg" viewBox="-7.7116809713429575 -16.75102391736263 313.60835950127966 385.58404856714714" role="img" preserveAspectRatio="xMidYMid meet">
    
 <g class ="stylesvgtop">
    <path d="M298.185 264.061l-149.093 88.021L0 264.061V88.021L149.092 0l149.093 88.021v176.04z"></path>  
    <text x="150" y="190" ><tspan>DATO</tspan></text>
</g>
</svg>
</a>
</div>

                    <div class="column">
                    <div class="ui inverted segment"><NavBar />
                    </div>
                    </div>
  </div>
    </div>
    </div>


  <div id="home" class="row">
         <Home />
  </div>

  <div id="product"class="row">
         <Product/>
  </div>

  <div id="about" class="row">
          <About />
  </div>


   <div id="featured" class="row">
           <Featured />
   </div>
 
   <div id="contact" class="row">
            <Contact />
   </div>

    <div id="demo" class="row">
             <Demo />
     </div>


</div>


{this.props.children}

</div>

 
     
    );
  }
}

export default App;




