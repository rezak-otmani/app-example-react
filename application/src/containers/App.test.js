import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { Provider } from "react-redux";
import store from '../redux/store.js';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Provider store={store}><App /></Provider>, div);
  ReactDOM.unmountComponentAtNode(div);
   const tree = renderer.create(
    <Provider store={store}><App /></Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
   

});
