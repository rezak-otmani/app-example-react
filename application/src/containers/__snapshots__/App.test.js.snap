// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="pusher"
>
  <div
    className="ui vertical masthead center aligned segment"
  >
    <div
      className="overlay fixed"
    >
      <div
        className="ui link list"
      >
        <a
          className="item"
          href="#"
          onClick={[Function]}
        >
          <button
            className="ui inverted grey mini circular icon button"
            title="Home"
          >
                   
          </button>
        </a>
        <br />
        <a
          className="item"
          href="#product"
          onClick={[Function]}
        >
          <button
            className="ui inverted grey mini circular icon button"
            title="Product"
          />
        </a>
        <br />
        <a
          className="item"
          href="#about"
          onClick={[Function]}
        >
          <button
            className="ui inverted grey mini circular icon button"
            title="About"
          />
        </a>
        <br />
        <a
          className="item"
          href="#featured"
          onClick={[Function]}
        >
          <button
            className="ui inverted grey mini circular icon button"
            title="Featured"
          />
        </a>
        <br />
        <a
          className="item"
          href="#contact"
          onClick={[Function]}
        >
          <button
            className="ui inverted grey mini circular icon button"
            title="Contact"
          />
        </a>
        <br />
        <a
          className="item"
          href="#demo"
          onClick={[Function]}
        >
          <button
            className="ui inverted grey mini circular icon button"
            title="Demo"
          />
        </a>
      </div>
    </div>
    <div
      className="row"
      id="home"
    >
      <div
        className="pusher"
      >
        <div
          className="ui inverted vertical masthead center aligned segment"
        >
          <div
            className="ui inverted vertical strip segment"
          >
            <div
              className="headertop"
            >
              <div
                className="ui three columns center aligned stackable grid"
              >
                <div
                  className="column"
                >
                  <a
                    href=""
                    tabIndex="-1"
                    target="_self"
                  >
                    <svg
                      className="logo"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="-7.7116809713429575 -16.75102391736263 313.60835950127966 385.58404856714714"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        className="stylesvgtop"
                      >
                        <path
                          d="M298.185 264.061l-149.093 88.021L0 264.061V88.021L149.092 0l149.093 88.021v176.04z"
                        />
                        <text
                          x="150"
                          y="190"
                        >
                          <tspan>
                            DATO
                          </tspan>
                        </text>
                      </g>
                    </svg>
                  </a>
                </div>
                <div
                  className="column"
                >
                  <div
                    className="ui inverted segment"
                  >
                    <div
                      className="ui inverted segment"
                    >
                      <div
                        className="ui secondary stackable massive compact menu"
                      >
                        <a
                          className="olive item active"
                          href="#"
                          onClick={[Function]}
                        >
                          Home
                        </a>
                        <a
                          className="item"
                          href="#product"
                          onClick={[Function]}
                        >
                          Product
                        </a>
                        <a
                          className="item"
                          href="#about"
                          onClick={[Function]}
                        >
                          About
                        </a>
                        <a
                          className="item"
                          href="#featured"
                          onClick={[Function]}
                        >
                          Featured
                        </a>
                        <a
                          className="item"
                          href="#contact"
                          onClick={[Function]}
                        >
                          Contact
                        </a>
                        <a
                          className="item"
                          href="#demo"
                          onClick={[Function]}
                        >
                          Demo
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="ui text container"
          >
            <h1
              className="ui huge inverted header"
            >
              <div
                className="ui two wide columns center aligned stackable grid"
              >
                EMPOWERING GROWTH WITH REAL DATA
              </div>
            </h1>
            <a
              className="header item"
              href="#product"
              onClick={[Function]}
            >
              <div
                className="ui image"
              >
                <div
                  className="ui small fade reveal image"
                >
                  <img
                    alt=""
                    className="visible content"
                    src="image_link1.png"
                  />
                  <img
                    alt=""
                    className="hidden content"
                    src="image_link2.png"
                  />
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
      id="product"
    >
      <div
        className="pusher"
      >
        <div
          className="ui vertical masthead center aligned segment"
        >
          <div
            className="center aligned column"
          >
            <div
              className="ui sizer vertical segment"
            >
              <h1
                className="ui huge header"
              >
                OUR TECHNOLOGY
              </h1>
            </div>
          </div>
          <div
            className="ui seven columns center aligned stackable grid"
          >
            <div
              className="column"
            >
              <h2
                className="ui left aligned header"
              >
                <div
                  className="content"
                >
                  <span
                    className="ui olive header"
                  >
                     01 /
                  </span>
                   FAST
                  <div
                    className="sub header"
                  >
                    I'm a paragraph. Click here to add your own text and edit me. It’s easy. Just click “Edit Text” or double click me to add your                                                                   content and make changes to the font. Feel free to drag and drop me a ywhere you like on your page.
                  </div>
                </div>
              </h2>
            </div>
            <div
              className="column"
            >
              <h2
                className="ui left aligned header"
              >
                <div
                  className="content"
                >
                  <span
                    className="ui olive header"
                  >
                      02 /
                  </span>
                   SECURE
                  <div
                    className="sub header"
                  >
                    I'm a paragraph. Click here to add your own text and edit me. It’s easy. Just click “Edit Text” or double click me to add your own                                                                           content and make changes to the font. Feel free to drag and drop me an ywhere you like on your page.
                  </div>
                </div>
              </h2>
            </div>
            <div
              className="column"
            >
              <h2
                className="ui left aligned header"
              >
                <div
                  className="content"
                >
                  <span
                    className="ui olive header"
                  >
                       03 /
                  </span>
                   EASY
                  <div
                    className="sub header"
                  >
                    I'm a paragraph. Click here to add your own text and edit me. It’s easy. Just click “Edit Text” or double click me to add your                                                                   own content and make changes to the font. Feel free to drag and drop me anywhere you like on your page.
                  </div>
                </div>
              </h2>
            </div>
          </div>
          <div
            className="ui three columns center aligned grid"
          >
            <div
              className="ui reveal"
            >
              <div
                className="visible content"
              >
                <img
                  alt=""
                  className="ui Big image"
                  src="graphics.png"
                />
              </div>
              <div
                className="hidden content"
              >
                <img
                  alt=""
                  className="move left"
                  src="ecran.jpg"
                />
              </div>
            </div>
          </div>
          <a
            className="header item"
            href="#demo"
            onClick={[Function]}
          >
            <div
              className="positive massive ui button"
            >
              Try a demo
              <i
                className="right arrow icon"
              />
            </div>
          </a>
          <br />
          <br />
          <br />
        </div>
      </div>
    </div>
    <div
      className="row"
      id="about"
    >
      <div
        className="pusher"
      >
        <div
          className="ui vertical olive inverted masthead center aligned segment"
        >
          <div
            className="center aligned column"
          >
            <div
              className="ui sizer vertical segment"
            >
              <h1
                className="ui huge header"
              >
                MEET DATO
              </h1>
            </div>
          </div>
          <div
            className="ui seven columns center aligned stackable grid"
          >
            <div
              className="column"
            >
              <img
                alt=""
                src="story.png"
              />
              <h2
                className="ui left aligned header"
              >
                <div
                  className="ui divider"
                />
                <div
                  className="content"
                >
                  OUR STORY
                  <div
                    className="sub header"
                  >
                    <br />
                    I'm a paragraph. Click here to add your own text and edit me. It’s easy. Just click “Edit Text” or double click me to add your own content and make changes to the                              font. Feel free to drag and drop me a ywhere you like on your page.
                  </div>
                </div>
              </h2>
            </div>
            <div
              className="column"
            >
              <img
                alt=""
                src="vision.png"
              />
              <h2
                className="ui left aligned header"
              >
                <div
                  className="ui divider"
                />
                <div
                  className="content"
                >
                  OUR VISION
                  <div
                    className="sub header"
                  >
                    <br />
                    I'm a paragraph. Click here to add your own text and edit me. It’s easy. Just click “Edt Text” or double click me to add your own content and make changes to                                   the font. Feel free to drag and drop me an ywhere you like on your page.
                  </div>
                </div>
              </h2>
            </div>
            <div
              className="column"
            >
              <img
                alt=""
                src="technology.png"
              />
              <h2
                className="ui left aligned header"
              >
                <div
                  className="ui divider"
                />
                <div
                  className="content"
                >
                  OUR TECHNOLOGY
                  <div
                    className="sub header"
                  >
                    <br />
                    I'm a paragraph. Click here to add your own text and edit me. It’s easy. Just click “Edit Text” or double clic me to add your own content and make changes to                                   the font. Feel free to drag and drop me anywhere you like on your page.
                  </div>
                </div>
              </h2>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
      id="featured"
    >
      <div
        className="pusher"
      >
        <div
          className="ui vertical masthead center aligned segment"
        >
          <div
            className="center aligned column"
          >
            <div
              className="ui sizer vertical segment"
            >
              <h1
                className="ui huge header"
              >
                USING DATO
              </h1>
            </div>
          </div>
          <div
            className="ui seven columns center aligned stackable grid"
          >
            <div
              className="column"
            >
              <img
                alt=""
                src="sat.png"
              />
            </div>
            <div
              className="column"
            >
              <img
                alt=""
                src="up.png"
              />
            </div>
            <div
              className="column"
            >
              <img
                alt=""
                src="tech2023.png"
              />
            </div>
            <div
              className="seven column row"
            >
              <div
                className="column"
              >
                <img
                  alt=""
                  src="le.png"
                />
              </div>
              <div
                className="column"
              >
                <img
                  alt=""
                  src="infodata.png"
                />
              </div>
              <div
                className="column"
              >
                <img
                  alt=""
                  src="info.png"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
      id="contact"
    >
      <div
        className="pusher"
      >
        <div
          className="ui vertical olive inverted masthead center aligned segment"
        >
          <div
            className="center aligned column"
          >
            <div
              className="ui sizer vertical segment"
            >
              <h1
                className="ui huge header"
              >
                GET IN TOUCH
              </h1>
            </div>
          </div>
          <div
            className="ui seven columns center aligned stackable divided equal height stackable grid"
          >
            <div
              className="three wide column"
            >
              <div
                className="ui link list"
              >
                <a
                  className="item"
                  href=""
                >
                  info@mysite.com 
                </a>
                <a
                  className="item"
                  href=""
                >
                  info@mysite.com
                </a>
              </div>
            </div>
            <div
              className="three wide column"
            >
              <div
                className="ui link list"
              >
                <a
                  className="item"
                  href=""
                >
                  Tel: 123-456-7890
                </a>
                <a
                  className="item"
                  href=""
                >
                  Fax: 123-456-7890
                </a>
              </div>
            </div>
            <div
              className="three wide column"
            >
              <div
                className="ui link list"
              >
                <p
                  className="item"
                >
                  500 Terry Francois Street
                </p>
                <p
                  className="item"
                >
                  San Francisco, CA 94158
                </p>
              </div>
            </div>
          </div>
          <div
            className="ui billboard test ad"
            data-text=""
          >
            <div>
              <div
                className={undefined}
                style={
                  Object {
                    "height": "100%",
                    "position": "absolute",
                    "width": "100%",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "bottom": 0,
                      "display": "inherit",
                      "height": "100vh",
                      "left": 0,
                      "position": "relative",
                      "right": 0,
                      "top": 0,
                      "width": "100vw",
                    }
                  }
                >
                  Loading map...
                </div>
              </div>
              <div />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
      id="demo"
    >
      <div
        className="pusher"
      >
        <div
          className="ui vertical masthead center aligned segment"
        >
          <div
            className="center aligned column"
          >
            <h1
              className="ui  header"
            >
              <div
                className="content"
              >
                REQUEST DEMO
                <div
                  className="sub header"
                >
                  <h2>
                     Please fill your contact details below:
                  </h2>
                </div>
              </div>
            </h1>
          </div>
          <div
            className="row"
          >
            <div
              className="ui container"
            >
              <div
                className="center aligned column"
              >
                <form
                  className="ui form"
                  onSubmit={[Function]}
                >
                  <div>
                    <input
                      name="name"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onDragStart={[Function]}
                      onDrop={[Function]}
                      onFocus={[Function]}
                      placeholder="Nom"
                      type="text"
                      value=""
                    />
                  </div>
                  <br />
                  <div>
                    <input
                      name="email"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onDragStart={[Function]}
                      onDrop={[Function]}
                      onFocus={[Function]}
                      placeholder="Email"
                      type="email"
                      value=""
                    />
                  </div>
                  <br />
                  <div>
                    <input
                      name="subject"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onDragStart={[Function]}
                      onDrop={[Function]}
                      onFocus={[Function]}
                      placeholder="Sujet"
                      type="text"
                      value=""
                    />
                  </div>
                  <br />
                  <div>
                    <input
                      name="phone"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onDragStart={[Function]}
                      onDrop={[Function]}
                      onFocus={[Function]}
                      placeholder="Téléphone"
                      type="text"
                      value=""
                    />
                  </div>
                  <br />
                  <div>
                    <input
                      name="adresse"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onDragStart={[Function]}
                      onDrop={[Function]}
                      onFocus={[Function]}
                      placeholder="Adresse"
                      type="text"
                      value=""
                    />
                  </div>
                  <br />
                  <div>
                    <button
                      className="massive ui button"
                      disabled={true}
                      onClick={[Function]}
                      type="button"
                    >
                      Clear Values
                    </button>
                    <button
                      className="positive massive ui button"
                      disabled={false}
                      type="submit"
                    >
                      Send
                      <i
                        className="right arrow icon"
                      />
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
          <br />
          <div
            className="ui vertical olive inverted footer segment"
          >
            <div
              className="ui seven columns center aligned stackable grid"
            >
              <div
                className="three wide column"
              >
                <a
                  href=""
                  tabIndex="-1"
                  target="_self"
                >
                  <svg
                    className="logo"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    viewBox="-7.7116809713429575 -16.75102391736263 313.60835950127966 385.58404856714714"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      className="stylesvgbottom"
                    >
                      <path
                        d="M298.185 264.061l-149.093 88.021L0 264.061V88.021L149.092 0l149.093 88.021v176.04z"
                      />
                      <text
                        x="150"
                        y="190"
                      >
                        <tspan>
                          DATO
                        </tspan>
                      </text>
                    </g>
                  </svg>
                </a>
              </div>
              <div
                className="two wide column"
              >
                <h2
                  className="ui left aligned header"
                >
                  <div
                    className="ui divider"
                  />
                  <div
                    className="content"
                  >
                    LINKS
                    <div
                      className="sub header"
                    >
                      <div
                        className="ui link list"
                      >
                        <br />
                        <a
                          className="item active"
                          href="#"
                          onClick={[Function]}
                        >
                          Home 
                        </a>
                        <a
                          className="item"
                          href="#product"
                          onClick={[Function]}
                        >
                          Product
                        </a>
                        <a
                          className="item"
                          href="#about"
                          onClick={[Function]}
                        >
                          About
                        </a>
                        <a
                          className="item"
                          href="#featured"
                          onClick={[Function]}
                        >
                          Featured
                        </a>
                        <a
                          className="item"
                          href="#contact"
                          onClick={[Function]}
                        >
                          Contact
                        </a>
                        <a
                          className="item"
                          href="#demo"
                          onClick={[Function]}
                        >
                          Demo
                        </a>
                      </div>
                    </div>
                  </div>
                </h2>
              </div>
              <div
                className="two wide column"
              >
                <h2
                  className="ui left aligned header"
                >
                  <div
                    className="ui divider"
                  />
                  <div
                    className="content"
                  >
                    ABOUT
                    <div
                      className="sub header"
                    >
                      <div
                        className="ui link list"
                      >
                        <br />
                        <a
                          className="item"
                          href=""
                        >
                          info@mysite.com 
                        </a>
                        <a
                          className="item"
                          href=""
                        >
                          info@mysite.com
                        </a>
                        <p
                          className="item"
                        >
                          Tel: 123-456-7890
                        </p>
                        <p
                          className="item"
                        >
                          500 Terry Francois Street
                        </p>
                        <p
                          className="item"
                        >
                          San Francisco, CA 94158
                        </p>
                      </div>
                    </div>
                  </div>
                </h2>
              </div>
              <div
                className="two wide column"
              >
                <h2
                  className="ui left aligned header"
                >
                  <div
                    className="ui divider"
                  />
                  <div
                    className="content"
                  >
                    SOCIAL
                    <div
                      className="sub header"
                    >
                      <br />
                      <button
                        className="ui circular facebook icon button"
                      >
                        <i
                          className="facebook icon"
                        />
                      </button>
                      <button
                        className="ui circular twitter icon button"
                      >
                        <i
                          className="twitter icon"
                        />
                      </button>
                      <button
                        className="ui circular linkedin icon button"
                      >
                        <i
                          className="linkedin icon"
                        />
                      </button>
                      <button
                        className="ui circular google plus icon button"
                      >
                        <i
                          className="google plus icon"
                        />
                      </button>
                    </div>
                  </div>
                </h2>
              </div>
            </div>
          </div>
          <div
            className="ui vertical wide left aligned segment"
          >
            © 2018 by Rezak. App created with ReactJS
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
